engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -784,10 +784,15 @@ public void monitorHeartbeat(boolean waitForRecovery)
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
-          if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
+          if (sca.lastHeartbeatMillis != -1 && (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis)) {
             if (!isApplicationIdle()) {
```

Comment:
I am not sure but according to me when the application starts at that time some operators may not be deployed instantly and may be waiting for some resources may be at that time container may miss a heartbeat but should not be killed. 
I need to check this theory will get back to you on this.